### PR TITLE
create new keytype

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
Handles #3129

### CODE REVIEW

File: Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/GenerateKeyFlow.swift [hunk 1]
```diff
@@ -55,7 +55,7 @@ class GenerateKeyFlow: AddKeyFlow {
             return false
         }
 
-        return OwnerKeyController.importKey(privateKey, name: name, isDrivedFromSeedPhrase: true)
+        return OwnerKeyController.importKey(privateKey, name: name, type: .deviceGenerated, isDrivedFromSeedPhrase: true)
```
Comment (biafra23) [inline Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/GenerateKeyFlow.swift]: Minor typo in `isDrivedFromSeedPhrase`. It should be `isDerivedFromSeedPhrase`.

File: Multisig/Cross-layer/Analytics/TrackingEvent.swift [hunk 2]
```diff
@@ -16,6 +16,7 @@ enum TrackingUserProperty: String, UserProperty {
     case numKeysWalletConnect = "num_keys_walletconnect" // string, number of WalletConnect keys, "0" on fresh install
     case numKeysLedgerNanoX = "num_keys_ledger_nano_x" // string, number of Ledger Nano X keys, "0" on fresh install
     case numKeysKeystone = "num_keys_keystone" // string, number of Keystone keys, "0" on fresh install
+    case numKeysWeb3Auth = "num_keys_web3auth" // string, number of Keystone keys, "0" on fresh install
```
Comment (biafra23) [inline Multisig/Cross-layer/Analytics/TrackingEvent.swift]: Accidentally copied comment.

File: Multisig/UI/Settings/OwnerKeyManagement/KeyAddedViewController.swift [hunk 3]
```diff
@@ -66,6 +66,7 @@ fileprivate extension KeyType {
         case .deviceGenerated: return "Generate Owner Key"
         case .keystone: return "Connect Keystone"
         case .walletConnect: return "Connect WalletConnect"
+        case .web3Auth: return "Login via Web2"
```
Comment (biafra23) [inline Multisig/UI/Settings/OwnerKeyManagement/KeyAddedViewController.swift]: Is that the wording from the design? Do we even have a design for this?

File: Multisig/Logic/Models/KeyInfo.swift [hunk 4]
```diff
@@ -37,6 +55,11 @@ extension KeyInfo {
         set { type = Int16(newValue.rawValue) }
     }
 
+    var web3AuthType: LoginType {
+        get { LoginType(rawValue: Int(loginType)) ?? .none }
```
Comment (biafra23) [inline Multisig/Logic/Models/KeyInfo.swift:59]: These two line don't compile for me. 

`/Users/biafra/safe-ios/Multisig/Logic/Models/KeyInfo.swift:59:39 Cannot find 'loginType' in scope`

File: Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/GenerateKeyFlow.swift (hunk 1 above)
Comment (elgatovital) [reply Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/GenerateKeyFlow.swift]: would you make this flag part of the .importedKey type declaration? Generated keys are always imported from seed phrase. It only applies to imported keys.

File: Multisig/Logic/Models/KeyInfo.swift [hunk 5]
```diff
@@ -18,6 +18,24 @@ enum KeyType: Int, CaseIterable {
     case walletConnect = 2
     case ledgerNanoX = 3
     case keystone = 4
+    case web3Auth = 5
+}
+
+enum LoginType: Int, CaseIterable {
```
Comment (elgatovital) [inline Multisig/Logic/Models/KeyInfo.swift]: naming is too generic and could be more specific -> Web3AuthLoginType

File: Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/GenerateKeyFlow.swift (hunk 1 above)
Comment (biafra23) [reply Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/GenerateKeyFlow.swift]: web3auth generated keys are not generated from a seed phrase.

Comment (elgatovital) [reply Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/GenerateKeyFlow.swift]: web3auth is a different key type

File: Multisig/UI/Settings/OwnerKeyManagement/KeyAddedViewController.swift (hunk 3 above)
Comment (MouazAlzahabi) [reply Multisig/UI/Settings/OwnerKeyManagement/KeyAddedViewController.swift]: No, we don't have design yet, I just add it for now and later we change once we have the design, however this case won't be triggered for now

File: Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/GenerateKeyFlow.swift (hunk 1 above)
Comment (MouazAlzahabi) [reply Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/GenerateKeyFlow.swift]: I agree, but I won't change that part now and will keep it for future refactor

**Review by elgatovital — APPROVED** [review]

**Review by biafra23 — APPROVED** [review]
LGTM 🚀

**Pull request merged**